contracts/validator-manager/tests/ValidatorMessagesTests.t.sol

```diff
@@ -23,13 +25,261 @@ contract ValidatorMessagesTest is Test {
+        invalidPacked2[1] = 0x01;
+        vm.expectRevert(
+            abi.encodeWithSelector(ValidatorMessages.InvalidCodecID.selector, uint32(1))
+        );
+        ValidatorMessages.unpackSubnetToL1ConversionMessage(invalidPacked2);
+    }
+
+    function testSubnetConversionMessageInvalidTypeID() public {
+        bytes memory packed =
+            ValidatorMessages.packSubnetToL1ConversionMessage(DEFAULT_SUBNET_CONVERSION_ID);
+        // Invalid message type
+        bytes memory invalidPacked3 = packed;
+        invalidPacked3[5] = 0x01;
+        vm.expectRevert(ValidatorMessages.InvalidMessageType.selector);
+        ValidatorMessages.unpackSubnetToL1ConversionMessage(invalidPacked3);
+    }
+
+    function testRegisterSubnetValidatorMessageInvalidBLSKey() public {
+        vm.expectRevert(ValidatorMessages.InvalidBLSPublicKey.selector);
+        // 47 bytes
+        bytes memory invalidBLSKey = bytes(
+            hex"3456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678"
+        );
+        ValidatorMessages.packRegisterL1ValidatorMessage(
+            ValidatorMessages.ValidationPeriod({
+                subnetID: DEFAULT_SUBNET_ID,
+                nodeID: DEFAULT_NODE_ID,
+                registrationExpiry: DEFAULT_EXPIRY,
+                blsPublicKey: invalidBLSKey,
+                remainingBalanceOwner: DEFAULT_P_CHAIN_OWNER,
+                disableOwner: DEFAULT_P_CHAIN_OWNER,
+                weight: DEFAULT_WEIGHT
+            })
+        );
+    }
+
+    function testRegisterSubnetValidatorMessageInvalidInputLength() public {
+        bytes memory packed = _getPackedRegisterL1ValidatorMessage();
+        // Invalid length
+        bytes memory invalidPacked = new bytes(packed.length - 1);
+        for (uint256 i = 0; i < packed.length - 1; i++) {
+            invalidPacked[i] = packed[i];
+        }
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                ValidatorMessages.InvalidMessageLength.selector, uint32(193), uint32(194)
+            )
+        );
+        ValidatorMessages.unpackRegisterL1ValidatorMessage(invalidPacked);
+
+        // Invalid codec ID
+        bytes memory invalidPacked2 = _getPackedRegisterL1ValidatorMessage();
+        invalidPacked2[1] = 0x01;
+        vm.expectRevert(
+            abi.encodeWithSelector(ValidatorMessages.InvalidCodecID.selector, uint32(1))
+        );
+        ValidatorMessages.unpackRegisterL1ValidatorMessage(invalidPacked2);
+
+        // Invalid message type
+        bytes memory invalidPacked3 = _getPackedRegisterL1ValidatorMessage();
+        invalidPacked3[5] = 0x00;
+        vm.expectRevert(ValidatorMessages.InvalidMessageType.selector);
+        ValidatorMessages.unpackRegisterL1ValidatorMessage(invalidPacked3);
+    }
+
+    function testRegisterSubnetValidatorMessageInvalidCodecID() public {
+        (, bytes memory packed) = ValidatorMessages.packRegisterL1ValidatorMessage(
```

Comment:
Do this test case and `testRegisterSubnetValidatorMessageInvalidTypeID` below  re-test functionality tested above? I agree that it would make sense to remove it from the above in that case to have a single test for single error state but we should be able to use `_getPackedRegisterL1ValidatorMessage()` as well